### PR TITLE
Make GTK4 the Default Version and Enable Adwaita

### PR DESCRIPTION
**Pull request by danyeaw**
This PR:

- Switches to making GTK4 the default version
- Enables GtkSourceView for CSS highlighting with GTK4
- Enables libadwaita with GTK4

### PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I have read, and I am following the [Contributor guide](https://github.com/gaphor/gaphor/blob/main/CONTRIBUTING.md)
- [X] I have read, and I understand the [Code of Conduct](https://github.com/gaphor/gaphor/blob/main/CODE_OF_CONDUCT.md)

### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [X] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?
GTK version 3 is the default

Issue Number: #618 

### What is the new behavior?
GTK version 4 is the default

### Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by danyeaw** [comment]
I took a swim trying to compile GTK4 on Ubuntu 18.04, and it will be very difficult. It requires gnutls >= 3.6.13, and I tried to rebuild it using dpkg-buildpackage. One of the patches with the 3.5 version was pushed upstream, and even if I don't apply that, I am still getting test errors during the build. I think we should leave the AppImage on GTK3 along with macOS, and upgrade everything else.

**Comment by amolenaar** [comment]
This looks really promising. I like the direction you're taking this.

It's failing on console tests again. I wonder what makes them crash this time.

File: _packaging/make-script.py [hunk 1]
```diff
@@ -32,6 +32,9 @@ def make_gaphor_script():
         # Check for and remove two semicolons in path
         file.write("os.environ['PATH'] = os.environ['PATH'].replace(';;', ';')\n")
 
+        # Load gaphor with GTK3, the frozen gaphor-exe is segfaulting with GTK4
+        file.write("os.environ['GAPHOR_USE_GTK'] = '3'\n")
```
Comment (amolenaar) [inline _packaging/make-script.py:36]: What is the effect of this line? From what I read from the changes, Windowsa and Linux are fully running GTK4. Or is this only fixing the packaging step?

**Review by amolenaar — APPROVED** [review]

**Comment by amolenaar** [comment]
Really nice. It looks really nice with `Adw.ApplicationWindow `.

File: _packaging/make-script.py (hunk 1 above)
Comment (danyeaw) [reply _packaging/make-script.py:36]: For packaging, I am having PyInstaller use GTK3 for all platforms (macOS, Linux AppImage, and Windows). I am getting a segfault with GTK4, so we'll have to dive in to this more. I think it is still worth switching the default and updating our flatpak builds to use GTK4 though.

Comment (amolenaar) [reply _packaging/make-script.py:36]: Thanks, that was not completely clear to me when I scanned the code.

This is a really nice PR.

**Pull request merged**